src/dbnode/storage/bootstrap/bootstrapper/readers.go

```diff
@@ -112,31 +111,17 @@ func enqueueReadersGroupedByBlockSize(
 	logger *zap.Logger,
 	span opentracing.Span,
 	nowFn clock.NowFn,
+	infoFilesByShard bootstrap.InfoFileResultsPerShard,
 ) {
 	// Group them by block size.
 	groupFn := NewShardTimeRangesTimeWindowGroups
 	groupedByBlockSize := groupFn(shardTimeRanges, blockSize)
 
-	// Cache info files by shard.
-	readInfoFilesResultsByShard := make(map[uint32][]fs.ReadInfoFileResult)
-
 	// Now enqueue across all shards by block size.
 	for _, group := range groupedByBlockSize {
 		readers := make(map[ShardID]ShardReaders, group.Ranges.Len())
 		for shard, tr := range group.Ranges.Iter() {
-			readInfoFilesResults, ok := readInfoFilesResultsByShard[shard]
-			if !ok {
-				start := nowFn()
-				logger.Debug("enqueue readers read info files start",
-					zap.Uint32("shard", shard))
-				readInfoFilesResults = fs.ReadInfoFiles(fsOpts.FilePathPrefix(),
-					ns.ID(), shard, fsOpts.InfoReaderBufferSize(),
-					fsOpts.DecodingOptions(), persist.FileSetFlushType)
-				logger.Debug("enqueue readers read info files done",
-					zap.Uint32("shard", shard),
-					zap.Duration("took", nowFn().Sub(start)))
-				readInfoFilesResultsByShard[shard] = readInfoFilesResults
-			}
+			readInfoFilesResults := infoFilesByShard[shard]
```

Comment:
This will need a sanity check and possibly fallback too

Reply:
Not saying the fallback isn't a good idea, but curious as to the scenario where we'd end up trying to read info files for a shard we haven't already loaded? The `RunState` object should be initialized with all shards owned by the host. So potentially if we try and read info files for a shard that's been peer bootstrapped?

Reply:
More for error scenarios where for some reason we haven't added the shard to the `RunState`; if this is an invalid scenario it may be worth emitting an invariant violation here instead?

Reply:
Yeah, I think that's probably the more accurate thing to do. We could make it fallback to fetch the info files from disk, but I think that means something unexpected has happened